src/main/java/com/sinthoras/visualprospecting/mixins/ifu/ItemOreFinderToolMixin.java

```diff
@@ -0,0 +1,64 @@
+package com.sinthoras.visualprospecting.mixins.ifu;
+
+import com.encraft.dz.items.*;
+import com.sinthoras.visualprospecting.*;
+import com.sinthoras.visualprospecting.database.*;
+import gregtech.api.enums.*;
+import gregtech.api.objects.*;
+import java.util.*;
+import java.util.stream.*;
+import net.minecraft.block.*;
+import net.minecraft.entity.*;
+import net.minecraft.entity.player.*;
+import net.minecraft.item.*;
+import net.minecraft.world.*;
+import org.spongepowered.asm.mixin.*;
+import org.spongepowered.asm.mixin.injection.*;
+import org.spongepowered.asm.mixin.injection.At.*;
+import org.spongepowered.asm.mixin.injection.callback.*;
+
+@Mixin(value = ItemOreFinderTool.class, remap = false)
+public class ItemOreFinderToolMixin extends Item {
+
+  @Shadow
+  private static int found;
+  
+  @Shadow
+  private static int MAX_DAMAGE;
+
+  @Inject(
+      method = "onUpdate",
+      at = @At(value = "INVOKE", target = "Lcom/encraft/dz/items/ItemOreFinderTool;shouldKeepLooking()Z", shift = Shift.AFTER, remap = false),
+      locals = LocalCapture.CAPTURE_FAILSOFT,
+      remap = true
+  )
+  public void onGtOreFound(
+      ItemStack itemstack, 
+      World world, Entity entity, 
+      int par4, boolean par5,
+      CallbackInfo ci, 
+      ItemStack searchItem, ItemData data, boolean vanilla, int id, 
+      double cur_x, double cur_y, double cur_z, 
+      int min_x, int min_y, int min_z, 
+      int max_x, int max_y, int max_z, 
+      boolean keepLooking, 
+      int z1, int x1, int y1, 
+      Block tBlock, int meta, ItemStack inWorld, ItemData dataInWorld, List<OrePrefixes> oreTypes
+  ) {
+    if (vanilla || world.isRemote || found < MAX_DAMAGE || !(entity instanceof EntityPlayer)) return;
+    
+    final short foundMaterialMetaItemSubId = (short) dataInWorld.mMaterial.mMaterial.mMetaItemSubID;
+    final List<OreVeinPosition> discoveredOreVeins = listAssociateVeins( foundMaterialMetaItemSubId, x1, z1, world);
```

Comment:
Match API var naming: `blockX`, `blockZ`

Reply:
Didn't check if mixin would complain, guess it's ordinal anyways. Will try.

Reply:
Although it might be better to leave it as is and make an intermediate variable, so it's not confusing to which local this is corresponding to.

Reply:
You may rename whatever you want. Indeed, mixins match it via ordinal.

You should keep the name, if it is helpful for the original method, but I would rename obfuscated variables and bad namings

Reply:
Renaming injected method parameter would make it harder to connect it to the original local variable we got it from. I'll rename method parameter I pass values in, so it's more clear. 

If you think it's necessary to clear meaning/naming at the begging I can create "redundant" local variables with better names right before I need to use them.